microsoft/testsuites/core/azure_image_standard.py

```diff
@@ -1690,7 +1727,47 @@ def verify_no_swap_on_osdisk(self, node: Node) -> None:
         # partition is created on a logical device, such as /dev/dm-5. In this case,
         # we need to get the real block device name.
         lsblk = node.tools[Lsblk]
-        os_disk = lsblk.find_disk_by_mountpoint("/")
+
+        # Try to find the OS disk by root mountpoint, with fallback for btrfs subvolumes
+        os_disk = None
+        try:
+            os_disk = lsblk.find_disk_by_mountpoint("/")
+        except LisaException as e:
+            if "Could not find disk with mountpoint /" in str(e):
+                # Find OS disk by checking which disk
+                # contains partitions mounted at common system paths
+                node.log.debug(
```

Comment:
Move the logic into a separate method, so it's easier to read.

Reply:
Done